cmd/find.go

```diff
@@ -0,0 +1,472 @@
+	printString := subArgsHelper(ctx.String("print"), filepath, fileContent)
+	printMsg(findMSG{
+		pathKey: printString,
+	})
+}
+
+func execFlag(ctx *cli.Context, filepath string, fileContent contentMessage) {
+	commandString := subArgsHelper(ctx.String("exec"), filepath, fileContent)
+	commandArgs := strings.Split(commandString, " ")
+
+	cmd := exec.Command(commandArgs[0], commandArgs[1:]...)
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	if err := cmd.Run(); err != nil {
+		console.Fatalln(err)
+		console.Fatalln()
+	}
+	console.Println(string(out.Bytes()))
+}
+
+func watchFlag(path string, ctx *cli.Context) {
+	params := watchParams{
+		recursive: true,
+		accountID: fmt.Sprintf("%d", time.Now().Unix()),
+		events:    []string{"put"},
+	}
+
+	pathnameParts := strings.SplitAfter(path, "/")
+	alias := strings.TrimSuffix(pathnameParts[0], "/")
+
+	_, e := getHostConfig(alias) // extract the hostname alias from the path name if present
+	noAlias := (e == nil)
+
+	clnt, content, err := url2Stat(path)
+	errorIf(err, "Unable to construct client")
+	fileContent := parseContent(content)
+
+	if noAlias {
+		//pass hostname (alias) to to watchHelper so that it can remove from pathnames returned from watch
+		watchHelper(clnt, params, ctx, alias, fileContent)
+	} else {
+		watchHelper(clnt, params, ctx, "", fileContent)
+	}
+}
+
+func olderFlag(createTime time.Time, pattern string) bool {
+	i, err := timeHelper(pattern)
+	now := time.Now()
+	if err != nil {
+		console.Println("Error parsing string", err)
+		console.Fatalln()
+	}
+
+	//find all time in which the time in which the object was just created is after the current time
+	t := time.Date(now.Year(), now.Month(), now.Day()-i, now.Hour(), now.Minute(), 0, 0, time.UTC)
+	return createTime.Before(t)
+}
+
+func newerFlag(createTime time.Time, pattern string) bool {
+	i, err := timeHelper(pattern)
+	now := time.Now()
+	if err != nil {
+		console.Println("Error parsing string", err)
+		console.Fatalln()
+	}
+
+	t := time.Date(now.Year(), now.Month(), now.Day()-i, now.Hour(), now.Minute(), 0, 0, time.UTC)
+	return createTime.After(t) || createTime.Equal(t)
+}
+
+func largerFlag(size int64, pattern string) bool {
+	i, err := humanize.ParseBytes(pattern)
+	if err != nil {
+		console.Println("Error parsing string", err)
+		console.Fatalln()
+	}
+
+	return int64(i) < size
+}
+
+func smallerFlag(size int64, pattern string) bool {
+	i, err := humanize.ParseBytes(pattern)
+	if err != nil {
+		console.Println("Error parsing string", err)
+		console.Fatalln()
+	}
+
+	return int64(i) > size
+}
+
+//return an error if we encounter an error
+func doFind(clnt Client, ctx *cli.Context) {
+	pathnameParts := strings.SplitAfter(ctx.Args().Get(0), "/")
+	alias := strings.TrimSuffix(pathnameParts[0], "/")
+	_, err := getHostConfig(alias)
+
+	for content := range clnt.List(true, false, DirNone) {
+		fileContent := parseContent(content)
+		filePath := fileContent.Key
+
+		//traversing in a object store not a file path
```

Comment:
You should read https://github.com/golang/go/wiki/CodeReviewComments this as well and apply it throughout this PR.